src/workflomics_benchmarker/cwltool_runtime_benchmark.py

```diff
@@ -398,6 +403,57 @@ def compute_technical_benchmarks(self,workflow_execution_information) -> List[di
 
         return technical_benchmarks
 
+    
+    def append_to_yaml_file(self, original_file_path):
+        yaml = YAML()
+        yaml.indent(mapping=3)
+        # Load the existing YAML data into an OrderedDict
+        with open(original_file_path, 'r') as file:
+            data = yaml.load(file)
+
+        if not isinstance(data, OrderedDict):
+            # If the YAML data is not already an OrderedDict, convert it
```

Comment:
Could you add some documentation? I'd like to know what this method is intended for, it's a bit unclear by looking at the code

Reply:
Also, why is there tool-specific code in there? If not possible to generalize, it might be useful to keep that part of the code contained separately so it could be more easily refactored later.

Reply:
I will add the documentation. The tool-specific code was added to support the 2 available scientific benchmarks, but I agree that the tool-benchmark dependencies should be annotated separately. I created an issue for it https://github.com/Workflomics/workflomics-benchmarker/issues/17.